tiny_dnn/core/framework/tensor_storage.h

```diff
@@ -68,6 +68,20 @@ static inline size_t product(Container &c) {
                            std::multiplies<size_t>());
 }
 
+template <typename C1, typename C2>
+static inline size_t compute_offset(const C1 &c1, const C2 &c2) {
```

Comment:
The inner loop calculates the accumulated product of the last N dimensions and the external loop uses just that value, mutliplies by a value from C1 and accumulates the sum.
If you do everything backwards, you can do everything in 1 loop, without a temporary array.